pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -234,7 +234,15 @@ func (g DefaultCacheKeyGenerator) QueryRequest(_ context.Context, tenantID strin
 }
 
 func (g DefaultCacheKeyGenerator) QueryRequestError(_ context.Context, tenantID string, r MetricsQueryRequest) string {
-	return fmt.Sprintf("EC:%s:%s:%d:%d:%d", tenantID, r.GetQuery(), r.GetStart(), r.GetEnd(), r.GetStep())
+	start := r.GetStart()
+	end := r.GetEnd()
+	if start == end && r.GetStep() == 0 {
```

Comment:
I know that we document that `start == end` for instant queries. Do we specify anywhere that `step == 0` for instant queries?